NewForce-Cohort-9/tabloid-full-stack-code-crusaders
# Df branch - Tag CRUD

#14 Create a Tag - https://trello.com/c/mgdqFJ21
#15 Delete a Tag - https://trello.com/c/W0HQkSsT
#16 Edit a Tag - https://trello.com/c/bmlXTqRN

**Description**
This PR implements an admin feature that allows users to add, edit, and delete tags in the Tag Management section of the TabloidFullStack application. For the time being, all users are treated as admins, with future stories planned to enforce user permissions.

**Type of Change**
Enhancement (non-breaking change that improves existing functionality)

Changes Made

1. Tag Management Page:
Created CRUD functions for Tag Management

2. Controller Methods:
Added a POST, PUT, and DELETE methods in the TagController.

3. Frontend Changes:
Created a new view for Tag Management CRUD operations.

**Testing Instructions**

1. Preparation:
- Ensure all your local changes are committed and pushed to your branch.
- Switch to the main branch and update it with the latest changes:
- Ensure the latest version of the main branch is pulled (git pull origin main).
- git fetch --a will give you access to my branch "df-branch"
- git checkout df-branch to switch to my branch
- If you don't see the most recent version of my files, try git pull origin df-branch
- If you haven't already created the Tabloid database, please do so using the provided SQL scripts.
- Use this https://github.com/NewForce-Cohort-9/tabloid-full-stack-code-crusaders/blob/main/SQL/01_Tabloid_Create_DB.sql to define the necessary tables and columns.
- Use this https://github.com/NewForce-Cohort-9/tabloid-full-stack-code-crusaders/blob/main/SQL/02_Tabloid_Seed_Data.sql to populate the database with initial data.

2. Launch the Application:
- After ensuring your environment is properly set up, launch the application by clicking the "Start with Debugger" (green play button) in your IDE.
- Log In:
- Open the application in your browser.
- Navigate to the Login page via the navbar and log in with the following credentials:
_Email: "foo@bar.com"
Password: Use any password of your choosing_.
- After logging in, you will be redirected to the homepage.

3. Navigate to Tag Management:
- From the homepage, click on the Tag Management link in the navbar.

4. Verify:
- Create a Tag "MightDeleteLater". After creation, you will be redirected to the Tag Management page.
- Edit the Tag "MightDeleteLater", change the name to "DeleteAfterEdit". After editing, you will be redirected to the Tag Management page.
- Delete the Tag "DeleteAfterEdit", confirm deletion. After deleting, you will be redirected to the Tag Management page.

**Checklist:**
